src/connection/sim.ts

```diff
@@ -471,19 +484,18 @@ export class SimulatorPlugin implements Connection {
         this.simPvs.put(simulatorInfo.simulator);
       }
 
-      if (publish && simulatorInfo.simulator !== undefined) {
-        if (simulatorInfo.initialValue !== undefined) {
+      if (simulatorInfo.initialValue !== undefined) {
```

Comment:
On line 510 you used this: `pvSimulator && pvSimulator.getValue()`
Why not use 
```js
simulatorInfo.initialValue && simulatorInfo.simulator && simulatorInfo.simulator.updateValue(simulatorInfo.initialValue);
```
or 
```js
if (simulatorInfo.initialValue && simulatorInfo.simulator)
{
    simulatorInfo.simulator.updateValue(simulatorInfo.initialValue);
}
```
here? Because you are especially interested in if they are *undefined* as opposed to just *falsy*?

Reply:
Sorry kind of long winded...

Reply:
Thanks, Tom, for taking the time to explain this! I actually wasn't familiar with how the `&&` construct worked in JS before looking at your code and it can be easy to get carried away with a *cool new feature* so wanted to check when it is actually a relevant tool.